dockerfiles/init/modules/codenvy/templates/codenvy.env.erb

```diff
@@ -53,7 +54,7 @@ JAVA_HTTPS_PROXY_PORT=-Dhttps.proxyPort=<%= @https_proxy_for_codenvy.gsub(/^http
 JAVA_NO_PROXY=-Dhttp.nonProxyHosts='<%= @no_proxy_for_codenvy.gsub(/^https?\:\/\//, '').gsub(/^www./,'').split(",").uniq.join("|") %>|'
 <% end -%>
 
-JAVA_OPTS=-Xms512m -Xmx<%= @codenvy_server_xmx %>m -server -XX:+HeapDumpOnOutOfMemoryError -XX:+UseConcMarkSweepGC -XX:+CMSParallelRemarkEnabled -XX:+UseCMSInitiatingOccupancyOnly -XX:+ScavengeBeforeFullGC -XX:+CMSScavengeBeforeRemark -Dche.local.conf.dir=${CHE_LOCAL_CONF_DIR} -Dauth.handler.default=<%= scope.lookupvar('codenvy::auth_handler_default') %> <% if ! @http_proxy_for_codenvy.empty? or ! @https_proxy_for_codenvy.empty? -%>$JAVA_HTTP_PROXY_SET<% end -%> <% if ! @http_proxy_for_codenvy.empty? -%>$JAVA_HTTP_PROXY_HOST $JAVA_HTTP_PROXY_PORT<% end -%> <% if ! @https_proxy_for_codenvy.empty? -%>$JAVA_HTTPS_PROXY_HOST $JAVA_HTTPS_PROXY_PORT<% end -%><%- if ! @no_proxy_for_codenvy.empty? -%> $JAVA_NO_PROXY<% end -%><% if @http_proxy_for_codenvy.include? '@' -%> $JAVA_HTTP_USER_NAME $JAVA_HTTP_USER_PASSWORD<% end -%><% if @https_proxy_for_codenvy.include? '@' -%> $JAVA_HTTPS_USER_NAME $JAVA_HTTPS_USER_PASSWORD<% end %>
+JAVA_OPTS=-Xms512m -Xmx<%= @codenvy_server_xmx %>m -server -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=${CODENVY_JAVA_DUMPS_DIR} -XX:+UseConcMarkSweepGC -XX:+CMSParallelRemarkEnabled -XX:+UseCMSInitiatingOccupancyOnly -XX:+ScavengeBeforeFullGC -XX:+CMSScavengeBeforeRemark -Dche.local.conf.dir=${CHE_LOCAL_CONF_DIR} -Dauth.handler.default=<%= scope.lookupvar('codenvy::auth_handler_default') %> <% if ! @http_proxy_for_codenvy.empty? or ! @https_proxy_for_codenvy.empty? -%>$JAVA_HTTP_PROXY_SET<% end -%> <% if ! @http_proxy_for_codenvy.empty? -%>$JAVA_HTTP_PROXY_HOST $JAVA_HTTP_PROXY_PORT<% end -%> <% if ! @https_proxy_for_codenvy.empty? -%>$JAVA_HTTPS_PROXY_HOST $JAVA_HTTPS_PROXY_PORT<% end -%><%- if ! @no_proxy_for_codenvy.empty? -%> $JAVA_NO_PROXY<% end -%><% if @http_proxy_for_codenvy.include? '@' -%> $JAVA_HTTP_USER_NAME $JAVA_HTTP_USER_PASSWORD<% end -%><% if @https_proxy_for_codenvy.include? '@' -%> $JAVA_HTTPS_USER_NAME $JAVA_HTTPS_USER_PASSWORD<% end %>
```

Comment:
I'd rather spit it into several declarations next time. Something like:
JAVA_OOM_OPTIONS=-XX:+Heap... -XX:HeapDumpPath
JAVA_GC_OPTIONS=-XX:+UseConcMa ... 
-JAVA_OPTS=-Xms512m -Xmx<%= @codenvy_server_xmx %>m -server -XX:+HeapDumpOnOutOfMemoryError -XX:+UseConcMarkSweepGC -XX:+CMSParallelRemarkEnabled 
.......
.......
JAVA_OPTS=$JAVA_OOM_OPTIONS $JAVA_GC_OPTIONS

@riuvshin @skabashnyuk WDYT?